lib/src/simulator.dart

```diff
@@ -249,18 +249,23 @@ abstract class Simulator {
   /// If there are no timestamps pending to execute, nothing will execute.
   static Future<void> tick() async {
     if (_injectedActions.isNotEmpty) {
-       // case 1 : ( the usual Rohd case )
-       //   The previous delta cycle did NOT do 'registerAction( _currentTimeStamp );'.
-       //   In that case, _pendingTimestamps[_currentTimestamp] is null so we will
-       //   add a new empty list, which will trigger a new delta cycle.
-       // case 2 :
-       //   The previous delta cycle DID do 'registerAction( _currentTimestamp );'.
-       //   In that case, there is *already* another tick scheduled for
-       //   _currentTimestamp, and the injected actions will get called in
-       //   the normal way.
-      _pendingTimestamps.putIfAbsent( _currentTimestamp , () => ListQueue() );
+      // case 1 : ( the usual Rohd case )
+      // The previous delta cycle did NOT do
+      // 'registerAction( _currentTimeStamp );'.
+      // In that case, _pendingTimestamps[_currentTimestamp] is null so we will
+      // add a new empty list, which will trigger a new delta cycle.
+      //
+      // case 2 :
+      // The previous delta cycle DID do 'registerAction( _currentTimestamp );'.
+      // In that case, there is *already* another tick scheduled for
+      // _currentTimestamp, and the injected actions will get called in
+      //  the normal way.
+      //
       // Either way, the end result is that a whole new tick gets scheduled for
       // _currentTimestamp and any outstanding injected actions get executed.
+
+      // ignore: unnecessary_lambdas
```

Comment:
I think this lint is saying you could have done `ListQueue.new` instead of `() => ListQueue()`, which seems fair but not a big deal.  Better to avoid the lint rather than waive, since it's probably a better way

Reply:
> Yes I think the lint is saying that, but I also read somewhere that while you can do that with an ordinary function you can’t do it with a constructor.I have to admit I didn’t do the experiment, and I just used the ignore statement

you're right that it used to be that way, though the minimum SDK version for ROHD supports the new constructor tearoffs (https://medium.com/dartlang/dart-2-15-7e7a598e508a)

i'd be ok leaving this as-is for now as a legacy way of doing things